core/templates/pages/exploration-player-page/learner-experience/conversation-skin.component.ts

```diff
@@ -1130,10 +1175,15 @@ export class ConversationSkinComponent {
             this.explorationActuallyStarted = true;
           }
         }
-        if (!this.explorationPlayerStateService.isInQuestionMode()) {
+        if (
+          !this.explorationPlayerStateService.isInQuestionMode() &&
+            !this.explorationPlayerStateService.isInDiagnosticTestPlayerMode()
+        ) {
           this.explorationPlayerStateService.onPlayerStateChange.emit(
             nextCard.getStateName());
-        } else {
+        } else if (
+          !this.explorationPlayerStateService.isInDiagnosticTestPlayerMode()
```

Comment:
I think these if conditions are getting too confusing.

Can we do something like this -- at init(), determine what mode this is in and store it in a variable that never gets edited again. Then all future decisions on what to do reference that variable.

Reply:
Yes, Sean, I understand these “If” conditions may cause confusion.
Currently, I am a little worried to refactor this in the current PR, because of the following reasons:
- The conversation skin component currently handles 3 different engines i.e., the exploration player engine, the question player engine, and the diagnostic test player engine.
- The refactor may cause unforeseen errors which can lead to delaying the current PR and project.
- I can create a new PR but for that, I have to take a look into the exploration-player engine service and how it is connected with the conversation skin component.

In my understanding adding explanatory comments above the “If” conditions may help.

Looking forward to your suggestion.

Thank you

Reply:
I think my concern is that I'm not convinced your code is correct, and it sounds like you're not convinced either since you haven't really looked into the exploration-player engine service. It also makes this whole thing into a "big ball of mud" since folks will be afraid to touch it.

Given this, I would still suggest doing the refactor. My suggestion would be to first do the refactor without the diagnostic test stuff -- just preserve the existing behaviour. Then add the diagnostic test stuff to it -- it will be simpler to do that after the refactor.

Reply:
Updated the logic by putting the modes that are expected in a particular block of code. Please take a look.